File: fixtures/java/src/com/acme/order/StockGuard.java
package com.acme.order;

/**
 * Throw-site regression case: a guard clause and a rethrow must each emit a
 * {@code throws} edge anchored at its own statement, and the {@code throws}
 * clause on {@code check} must stay a plain reference (propagation, not a
 * failure exit).
 */
public class StockGuard {

    void ensure(int quantity) {
        if (quantity < 0) {
            throw new EmptyBasketException("negative quantity");
        }
        try {
            check(quantity);
        } catch (EmptyBasketException e) {
            throw e;
        }
    }

    void check(int quantity) throws EmptyBasketException {
    }
}

File: fixtures/csharp/src/Acme/Order/StockGuard.cs
namespace Acme.Order;

/// <summary>
/// Throw-site case: a guard clause and a rethrow must each emit a `throws`
/// edge anchored at its own statement.
/// </summary>
public class StockGuard
{
    internal void Ensure(int quantity)
    {
        if (quantity < 0)
        {
            throw new EmptyBasketException("negative quantity");
        }
        try
        {
            Check(quantity);
        }
        catch (EmptyBasketException e)
        {
            throw e;
        }
    }

    internal void Check(int quantity)
    {
    }
}
